src/legacy/ui/public/inspector/inspector.tsx

```diff
@@ -72,10 +73,14 @@ function open(adapters: Adapters, options: InspectorOptions = {}): InspectorSess
       if an inspector can be shown.`);
   }
 
-  return openFlyout(<InspectorPanel views={views} adapters={adapters} title={options.title} />, {
-    'data-test-subj': 'inspectorPanel',
-    closeButtonAriaLabel: closeButtonLabel,
-  });
+  // TODO: Remove the null assumption as soon as we have a guarantee core is always ready
+  return getNewPlatform().start.core!.overlays.openFlyout(
```

Comment:
cc @epixa that is the code where we want to remove the exclamation mark once the `new_platform` in the legacy world properly knows `core` cannot be `null` at that point, as discussed in earlier meetings.